receiver/hostmetricsreceiver/internal/scraper/processesscraper/processes_scraper.go

```diff
@@ -50,7 +87,54 @@ func (s *scraper) start(context.Context, component.Host) error {
 }
 
 func (s *scraper) scrape(_ context.Context) (pdata.MetricSlice, error) {
+	now := pdata.TimestampFromTime(time.Now())
+
 	metrics := pdata.NewMetricSlice()
-	err := appendSystemSpecificProcessesMetrics(metrics, s.startTime, s.misc)
+	metrics.EnsureCapacity(metricsLength)
+
+	processMetadata, err := s.getProcessesMetadata()
+	if err != nil {
+		err = scrapererror.NewPartialScrapeError(err, metricsLength)
+	}
+
+	if enableProcessesCount && processMetadata.countByStatus != nil {
+		setProcessesCountMetric(metrics.AppendEmpty(), s.startTime, now, processMetadata.countByStatus)
+	}
+
+	if enableProcessesCreated && processMetadata.processesCreated != nil {
+		setProcessesCreatedMetric(metrics.AppendEmpty(), s.startTime, now, *processMetadata.processesCreated)
+	}
+
 	return metrics, err
 }
+
+func setProcessesCountMetric(metric pdata.Metric, startTime, now pdata.Timestamp, countByStatus map[string]int64) {
+	metadata.Metrics.SystemProcessesCount.Init(metric)
+	ddps := metric.Sum().DataPoints()
+
+	// sort so that we add datapoints in a deterministic ordering
+	sortedStatuses := make([]string, 0, len(countByStatus))
```

Comment:
Is a deterministic ordering actually needed, or is this just for testing?  If the latter, we should do the sorting in the test, rather than in code.

Reply:
Addressed as part of #5536

Reply:
Moved sorting to test.